### PR TITLE
Docs: fix 50+ errors in documentation

### PR DESCRIPTION
**Pull request by Aman-Codes**
* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

Fixed 50+ errors in documentation

### CODE REVIEW

**Review by squat — COMMENTED** [review]
Great work and thanks for the improvements! I just have one suggestion and then we can merge!

File: docs/components/store.md [hunk 1]
```diff
@@ -258,7 +258,7 @@ Thanos Store Gateway supports an index cache to speed up postings and series loo
 
 The `in-memory` index cache is enabled by default and its max size can be configured through the flag `--index-cache-size`.
 
-Alternatively, the `in-memory` index cache can also by configured using `--index-cache.config-file` to reference to the configuration file or `--index-cache.config` to put yaml config directly:
+Alternatively, the `in-memory` index cache can also be configured using `--index-cache.config-file` to refer to the configuration file or `--index-cache.config` to put yaml config directly:
```
Comment (squat) [inline docs/components/store.md:261]:
```suggestion
Alternatively, the `in-memory` index cache can also be configured using `--index-cache.config-file` to reference the configuration file or `--index-cache.config` to put yaml config directly:
```

**Comment by Aman-Codes** [comment]
@squat Updated it

**Review by squat — APPROVED** [review]
Thanks!!!!! 🚀🚀🚀

**Comment by Aman-Codes** [comment]
The Circleci test is failing and I am not able to figure out the reason from the error logs.
@squat Could you please guide me if I need to change something from my end?

**Comment by squat** [comment]
It's a flaky test it seems. I'm rerunning it

**Pull request merged**